hal/src/main/native/athena/HAL.cpp

```diff
@@ -353,6 +355,38 @@ size_t HAL_GetComments(char* buffer, size_t size) {
   return toCopy;
 }
 
+void InitializeTeamNumber(void) {
+  char hostnameBuf[25];
+  auto status = gethostname(hostnameBuf, sizeof(hostnameBuf));
+  if (status != 0) {
+    teamNumber = 0;
+    return;
+  }
+
+  std::string hostname{hostnameBuf, sizeof(hostnameBuf)};
+
+  auto start = hostname.find('-') + 1;
+  auto end = hostname.find('-', start);
+
+  if (start == std::string::npos || end == std::string::npos) {
+    teamNumber = 0;
+    return;
+  }
+
+  try {
+    teamNumber = std::stoi(hostname.substr(start, end - start));
```

Comment:
Use `wpi::parse_integer` (in StringExtras.h) instead.  This works with std::string_view.  Also StringExtras has some tools for dealing with the '-' splitting.

Reply:
A regular expression would also make extracting the team number trivial.

Reply:
I think the simple search for '-' is sufficient and likely faster than dealing with a full regex here.